platforms/access-bridge/src/services/stripe-service.ts

```diff
@@ -57,4 +60,40 @@ export class StripeService {
 
     return productsWithPrices;
   }
+
+  /**
+   * Creates a Stripe Checkout session based on the provided price ID.
+   * @param viewer Email address and viewer id from the auth token used for creating the checkout session.
+   * @param params Stripe checkout params to use for creating the checkout session.
+   * @returns A Promise resolving to a Stripe Checkout Session object, including a URL for the checkout page.
+   */
+  async createCheckoutSession(viewer: Viewer, params: StripeCheckoutParams): Promise<Stripe.Checkout.Session> {
+    const sessionParams: Stripe.Checkout.SessionCreateParams = {
+      payment_method_types: ['card'],
+      line_items: [
+        {
+          price: params.price_id,
+          quantity: 1,
+        },
+      ],
+      metadata: {
+        viewer_id: viewer.id,
+      },
+      customer_email: viewer.email,
+      mode: params.mode,
+      success_url: params.success_url,
+      cancel_url: params.cancel_url,
+
+      // Conditionally include `subscription_data` only if mode is `subscription`
+      ...(params.mode === 'subscription' && {
```

Comment:
What is mode used for besides 'subscription'?

Reply:
Mode can be also `payment` (for a single payment and not a subscription), not that we support it atm, but it will be eventually.